glide-core/redis-rs/redis/src/cluster_async/mod.rs

```diff
@@ -2337,6 +2453,58 @@ where
         }
     }
 
+    // The function is used to send an AUTH command to a node and then retry the original request.
+    // The function is used when the original request failed due to an AUTH error.
+    // Cases in which the function is helpful:
+    // 1. The password was changed, the connection was not re-established, and the request failed due to an AUTH error.
+    // 2. The protection method is allowing X hours of connection without authentication, and the request failed due to an AUTH error.
+    async fn send_re_auth_request(
```

Comment:
the function name doesn't clarify what it actually does - please rename to re_auth_and_retry_request, or something similar. Reading the function name alone I though it only sends re-auth